LayoutTests/http/tests/links/redirect-301-clear-memory-cache.html

```diff
@@ -0,0 +1,17 @@
+<html>
+<head>
+<title>301 Redirect</title>
+<script>
+  function runTest() {
+    if (window.testRunner) {
+        testRunner.waitUntilDone();
+    }
+    window.setTimeout(function() {
+      internals.clearMemoryCache();
+      window.location = 'resources/redirect-helper.pl?301#key=foo';
```

Comment:
I am not sure to follow how this test is supposed to work.
The issue is that we reuse resources that were previously cached.
This test should first:
- cache a resource with #key=bar.
- load the same resource with #key=foo
- check we do not reuse the #key=bar resource.

I do not see the first step, I would expect this test to pass with and without the patch.
Can you check this?

Reply:
This test case does work as we did also have a bug with when we had no fragments, but I'll add a test case to handle this scenario.